sdk/core/Azure.Core/src/Serialization/ModelWriter.SequenceBuilder.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core.Serialization
+{
+    internal sealed partial class ModelWriter : IDisposable
+    {
+        private sealed class SequenceBuilder : IBufferWriter<byte>, IDisposable
+        {
+            private struct Buffer
+            {
+                public byte[] Array;
+                public int Written;
+            }
+
+            private volatile Buffer[] _buffers; // this is an array so items can be accessed by ref
+            private volatile int _count;
+            private int _segmentSize;
+
+            /// <summary>
+            /// Initializes a new instance of <see cref="SequenceBuilder"/>.
+            /// </summary>
+            /// <param name="segmentSize">The size of each buffer segment.</param>
+            public SequenceBuilder(int segmentSize = 16384)
```

Comment:
Would it be possible to document why this constant was chosen?

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/38137

Reply:
TY!